### PR TITLE
Completely remove instanceof from summary framework

### PR DESCRIPTION
**Pull request by rguo-aws**
*Issue #, if available:*
#212 
*Description of changes:*
Completely remove instanceof from summary framework

*Tests:*
tested on docker

*Code coverage percentage for this patch:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by aditjind** [comment]
LGTM!

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotClusterSummary.java [hunk 1]
```diff
@@ -128,6 +131,30 @@ public JsonElement toJson() {
...
 
+  @Override
+  public List<GenericSummary> getNestedSummaryList() {
+    return new ArrayList<>(hotNodeSummaryList);
+  }
+
+  @Override
+  public GenericSummary buildNestedSummary(String summaryTable, Record record) {
+    GenericSummary ret = null;
+    if (summaryTable.equals(HotNodeSummary.HOT_NODE_SUMMARY_TABLE)) {
```
[the remark this answers is not in the document]
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotClusterSummary.java:142]: good catch. let me add that

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotResourceSummary.java [hunk 2]
```diff
@@ -59,6 +60,7 @@
   private double maxValue;
   private int timePeriod;
   private String metaData;
+  private final List<TopConsumerSummary> topConsumerSummaryList;
```
[the remark this answers is not in the document]
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotResourceSummary.java]: removed final

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotNodeSummary.java [hunk 3]
```diff
@@ -157,6 +166,41 @@ public JsonElement toJson() {
     return summaryObj;
   }
 
+  @Override
+  public List<GenericSummary> getNestedSummaryList() {
+    List<GenericSummary> summaries = new ArrayList<>();
+    summaries.addAll(hotResourceSummaryList);
+    summaries.addAll(hotShardSummaryList);
```
[the remark this answers is not in the document]
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/HotNodeSummary.java:179]: thanks. I have added a comment above this method

**Pull request merged**